### PR TITLE
Fixes duplicate Brave Ads ad history entries are missing

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave/brave-browser/issues/7869

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

NOTES: An ad will need to be displayed upon launch before the history is updated to reflect the fix. Legacy ads history will show duplicates, i.e. viewed and clicked actions

- View multiple ads (and on occasion click some ads)
- Confirm ads which were clicked only appeared as clicked and do not appear as Viewed in the ad's history
- Confirm duplicate ads are shown
- Confirm ads are shown across multiple days
- Confirm ads older than 7 days are not shown

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc [hunk 1]
```diff
@@ -1240,17 +1241,30 @@ void LedgerImpl::ConfirmAd(const std::string& info) {
...
     case ads::ConfirmationType::DOWNVOTE: {
-      notification_info->type = confirmations::ConfirmationType::DOWNVOTE;
+      // Should never be reached as DOWNVOTE is handled by ConfirmAction,
+      // refactor code to abstract different ConfirmationTypes
+      NOTREACHED();
+      break;
+    }
+
+    case ads::ConfirmationType::CONVERSION: {
+      // Should never be reached as CONVERSION is handled by ConfirmAction,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: Unrelated change however I had to add this `case` statement due to other required changes

**Comment by tmancey** [comment]
CI failed for unrelated reasons

**Comment by bsclifton** [comment]
Rebased so that we'd get a fresh CI run 😄 Last one (after force push) failed with version mismatch. Will keep an eye on this

**Comment by bbondy** [comment]
Per DM with Terry, merging since Nejc is codeowner but he's away.
